signer/src/storage/postgres.rs

```diff
@@ -490,6 +490,56 @@ impl super::DbRead for PgStore {
         .map_err(Error::SqlxQuery)
     }
 
+    /// Get the ancestral bitcoin blocks for the given block hash.
+    async fn get_ancestral_bitcoin_blocks(
+        &self,
+        block_hash: &model::BitcoinBlockHash,
+        limit: u16,
+    ) -> Result<Vec<model::BitcoinBlock>, Error> {
+        sqlx::query_as::<_, model::BitcoinBlock>(
+            r#"
+            WITH RECURSIVE context_window AS (
+                -- Anchor member: Initialize the recursion with the chain tip
+                SELECT block_hash, block_height, parent_hash, confirms, 1 AS depth
+                FROM sbtc_signer.bitcoin_blocks
+                WHERE block_hash = $1
+
+                UNION ALL
+
+                -- Recursive member: Fetch the parent block using the last block's parent_hash
+                SELECT parent.block_hash, parent.block_height, parent.parent_hash,
+                        parent.confirms, last.depth + 1
```

Comment:
nit: this should follow the same style like we do for our other queries here, like:
```sql
SELECT 
    block_hash
  , parent_hash
  , confirms
  , 1 AS depth
FROM ...
```

Reply:
Will do

Reply:
Just FYI incoming in #710 is a table-valued function that will let you remove the explicit recursive query.